### ISSUE TITLE
Feature make scorecard brew formula

### ISSUE THREAD

**Issue by developer-guy** [ossf/scorecard#1364]
**Is your feature request related to a problem? Please describe.**
This is not related to a problem.

**Describe the solution you'd like**

scorecard project already creates releases by using the GoReleaser project, so, GoReleaser has already support for creating brew formulas for free[^1]. The thing that needs to be done is just create a repository for storing these formulas called `tap`.

**Describe alternatives you've considered**

**Additional context**

* https://appliedgo.net/release2/

[^1]: https://goreleaser.com/customization/homebrew/

cc: @naveensrinivasan @dentrax

**Comment by naveensrinivasan** [comment]
I agree, this would be a great addition. But I don't think we have time now to maintain another repository and a project. 
@azeemshaikh38 @laurentsimon Thoughts

**Comment by laurentsimon** [comment]
agreed. We d'love to have it, but we don't have the resources to take it on ourselves right now. 
Thanks for the suggestion @developer-guy

**Comment by developer-guy** [comment]
There is no need to maintain a repository btw because it will only hold these formulas. So, create a repository, and name it as `homebrew-tap`, that's all we need to do. Then, all these formulas will be created by the GoReleaser and pushed to that repository.

@naveensrinivasan @laurentsimon

**Comment by laurentsimon** [comment]
got for it!

cc @haydentherapper

**Comment by developer-guy** [comment]
just place it under ossf organization, that's all I need IMHO.

**Comment by developer-guy** [comment]
once you do this, I'll add the necessary section for releasing HomeBrew formulas with GoReleaser.

**Comment by naveensrinivasan** [comment]
The issue is that once we create the repo, there are going to be questions and issues which is something we have to take it up. That is my apprehension.

**Comment by developer-guy** [comment]
I don't think so until we change the way we work with this repository, because this repository is going to be maintained by the GoReleaser only.

**Comment by naveensrinivasan** [comment]
> I don't think so until we change the way we work with this repository, because this repository is going to be maintained by the GoReleaser only.

I agree, but there are going to be users who are going to ask for some support. The way in which we integrate goreleaser we might break that integration. We need to test it part of every release. All of these are kinds of chores.

**Comment by developer-guy** [comment]
I can maintain this repository if you want 🤩

**Comment by azeemshaikh38** [comment]
> I can maintain this repository if you want 🤩

That's very generous of you @developer-guy! Thank you for offering :) I would be very happy to have this feature for Scorecard. I vote that we go ahead with this. @laurentsimon @naveensrinivasan wdyt?

**Comment by laurentsimon** [comment]
Go for it, yeah! So we just need to create a `ossf/scorecard-tap` repo?

**Comment by developer-guy** [comment]
it makes sense, I think there is no restriction about the naming, but I'm not sure about that.

**Comment by naveensrinivasan** [comment]
Let me know the repo name, I can create one.

**Comment by developer-guy** [comment]
I think it must be in the form of `homebrew-<something>.`

**Comment by developer-guy** [comment]
like https://github.com/sigstore/homebrew-tap

**Comment by naveensrinivasan** [comment]
How about `homebrew-scorecard-tap`?

**Comment by azeemshaikh38** [comment]
Or `scorecard-homebrew-tap`? Since we already have `scorecard-actions`, it will seem more like a standard/follows a convention.

**Comment by developer-guy** [comment]
I think it must start with the `homebrew-` prefix, and the rest does not matter.

> On GitHub, your repository must be named homebrew-something to use the one-argument form of brew tap. The prefix “homebrew-“ is not optional. (The two-argument form doesn’t have this limitation, but it forces you to give the full URL explicitly.)

based on: https://docs.brew.sh/Taps#repository-naming-conventions-and-assumptions

**Comment by azeemshaikh38** [comment]
I see. Then, maybe `homebrew-scorecard`? With this, users should be able to invoke `homebrew tap ossf/scorecard`, right?

**Comment by developer-guy** [comment]
Users might be able to install for example scorecard like the following way if we name this repository `homebrew-scorecard`:

```shell
brew tap ossf/scorecard
brew install scorecard
```

but if we name it like this `homebrew-tap`, the command will become this:

```shell
brew tap ossf/tap
brew install scorecard
```

**Comment by naveensrinivasan** [comment]
What if ossf wants to support other brew packages? We need to think about that.

**Comment by azeemshaikh38** [comment]
> Users might be able to install for example scorecard like the following way if we name this repository `homebrew-scorecard`

That sounds great. My vote is for this option since its very intuitive and provides a better developer experience. 

> What if ossf wants to support other brew packages? We need to think about that.

Those repos will be called `homebrew-*`. For example, `homebrew-allstar`. Wdyt?

**Comment by developer-guy** [comment]
Nope, there must be one repository for the organization to include all the formulas for projects maintained by the organization. So, we should create one repository, as I mentioned above, like `homebrew-tap`, we shouldn't use the tool name within the repository name.

**Comment by developer-guy** [comment]
I think the most suitable name for this is `ossf/homebrew-tap` btw.

**Comment by developer-guy** [comment]
tap and formulas are two different things, tap means a repository for holding formulas, think that way.

**Comment by naveensrinivasan** [comment]
My vote is `homebrew-tap`.

**Comment by azeemshaikh38** [comment]
> tap and formulas are two different things, tap means a repository for holding formulas, think that way.

I see, didn't realize this was an org-wide thing. Ok, I also vote for `homebrew-tap` then.

**Comment by naveensrinivasan** [comment]
https://github.com/ossf/homebrew-tap created. @developer-guy I have sent the invite to you for the repository.

**Comment by azeemshaikh38** [comment]
Thanks Naveen!

**Comment by developer-guy** [comment]
Let's give try through the demo repository
* https://github.com/developer-guy/scorecard/runs/4449079235?check_suite_focus=true
* https://github.com/developer-guy/homebrew-tap

**Comment by developer-guy** [comment]
you can run:

```shell
$ brew tap developer-guy/tap
$ brew install scorecard
```

**Comment by developer-guy** [comment]
![Screen Shot 2021-12-07 at 23 41 18](https://user-images.githubusercontent.com/16693043/145103254-2e57dd40-7da3-49bd-affd-a91f6f148405.png)
![Screen Shot 2021-12-07 at 23 43 25](https://user-images.githubusercontent.com/16693043/145103330-d31776da-6119-43b5-8c78-1586fb8890f3.png)
![Screen Shot 2021-12-07 at 23 43 39](https://user-images.githubusercontent.com/16693043/145103337-5cd607d1-d287-41b5-9fb4-009f3353c700.png)

**Comment by developer-guy** [comment]
yay! 🥳 @naveensrinivasan @azeemshaikh38

**Comment by naveensrinivasan** [comment]
Cool do a PR! But what permission does the action needs to write to the other repository?

**Comment by developer-guy** [comment]
just GitHub token IMHO, I used my own PAT to do that.

**Comment by naveensrinivasan** [comment]
closing it because we have scorecard brew packages.
